scripts/deploy/build-deploy-common.sh

```diff
@@ -223,18 +231,23 @@ build_image() {
 
     local BUILD_ARGS_CMD=''
     local LABEL_PARAMETER=''
-
+    local DISABLE_CACHE_PARAMETER=''
 
     if is_pr_or_mr_build; then
         LABEL_PARAMETER=$(get_expiry_label_parameter)
     fi
 
+    if _disable_build_cache || ([[ "$(get_container_engine_cmd)" == "podman" ]] && _podman_version_under_4_5_0); then
+        DISABLE_CACHE_PARAMETER='--no-cache'
```

Comment:
A warning for the end user:

```suggestion
        echo 'Old version of Podman detected. Build cache disabled.' >&2
        DISABLE_CACHE_PARAMETER='--no-cache'
```

The "Old" could be replaced with "Unsupported". Not sure which working is more correct, though.